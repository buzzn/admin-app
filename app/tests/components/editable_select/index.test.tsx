import React from 'react';
import renderer from 'react-test-renderer';
import EditableSelect from 'components/editable_select';

describe('editable select tests', () => {
  const props = {
    field: { enum: ['first', 'second', 'third'] },
    editMode: true,
    input: { name: 'select', value: 'second' },
    meta: { active: false, dirty: false, error: '', touched: false },
    prefix: 'objPrefix',
  };

  test('basic view and edit rendering', () => {
    const noVal = renderer
      .create(<EditableSelect {...{ ...props, editMode: false, input: { name: 'select' } }} />)
      .toJSON();
    expect(noVal).toMatchSnapshot();

    const view = renderer.create(<EditableSelect {...{ ...props, editMode: false }} />).toJSON();
    expect(view).toMatchSnapshot();

    const edit = renderer.create(<EditableSelect {...props} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('raw value output', () => {
    const view = renderer.create(<EditableSelect {...{ ...props, editMode: false, withValue: true }} />).toJSON();
    expect(view).toMatchSnapshot();

    const edit = renderer.create(<EditableSelect {...{ ...props, withValue: true }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('active field', () => {
    const edit = renderer.create(<EditableSelect {...{ ...props, meta: { active: true } }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('with error', () => {
    const edit = renderer
      .create(<EditableSelect {...{ ...props, meta: { dirty: true, error: 'error', touched: true } }} />)
      .toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('with default value', () => {
    const edit = renderer
      .create(<EditableSelect {...{ ...props, defaultValue: { value: 'default', label: 'Default' } }} />)
      .toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('list override', () => {
    const edit = renderer
      .create(
        <EditableSelect
          {...{
            ...props,
            listOverride: [{ value: 'override1', label: 'Override1' }, { value: 'override2', label: 'Override2' }],
          }}
        />,
      )
      .toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('no default value', () => {
    const edit = renderer.create(<EditableSelect {...{ ...props, noDefault: true }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('no value translations', () => {
    const edit = renderer.create(<EditableSelect {...{ ...props, noValTranslations: true }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('override data', () => {
    const edit = renderer
      .create(
        <EditableSelect {...{ ...props, overrideData: { override: 'Override' }, input: { name: 'admin.override' } }} />,
      )
      .toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('prefix', () => {
    const edit = renderer.create(<EditableSelect {...{ ...props, prefix: 'customPrefix' }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('values prefix', () => {
    const edit = renderer.create(<EditableSelect {...{ ...props, valuesPrefix: 'customPrefix' }} />).toJSON();
    expect(edit).toMatchSnapshot();
  });

  test('with label', () => {
    const edit = renderer
      .create(<EditableSelect {...{ ...props, withLabel: true, input: { name: 'select', value: '' } }} />)
      .toJSON();
    expect(edit).toMatchSnapshot();

    const active = renderer
      .create(<EditableSelect {...{ ...props, withLabel: true, meta: { active: true } }} />)
      .toJSON();
    expect(active).toMatchSnapshot();
  });
});
