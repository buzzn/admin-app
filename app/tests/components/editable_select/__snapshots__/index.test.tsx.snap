// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editable select tests active field 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select undefined"
>
  <select
    className="custom-select form-control undefined undefined"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests basic view and edit rendering 1`] = `<span />`;

exports[`editable select tests basic view and edit rendering 2`] = `
<span>
  objPrefix.second
</span>
`;

exports[`editable select tests basic view and edit rendering 3`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests list override 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="override1"
    >
      objPrefix.override1
    </option>
    <option
      value="override2"
    >
      objPrefix.override2
    </option>
  </select>
</div>
`;

exports[`editable select tests no default value 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests no value translations 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      first
    </option>
    <option
      value="second"
    >
      second
    </option>
    <option
      value="third"
    >
      third
    </option>
  </select>
</div>
`;

exports[`editable select tests override data 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <input
    className="form-control"
    disabled={true}
    type="text"
    value="Override"
  />
</div>
`;

exports[`editable select tests prefix 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      customPrefix.first
    </option>
    <option
      value="second"
    >
      customPrefix.second
    </option>
    <option
      value="third"
    >
      customPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests raw value output 1`] = `
<span>
  second - objPrefix.second
</span>
`;

exports[`editable select tests raw value output 2`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      first - objPrefix.first
    </option>
    <option
      value="second"
    >
      second - objPrefix.second
    </option>
    <option
      value="third"
    >
      third - objPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests values prefix 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      customPrefix.first
    </option>
    <option
      value="second"
    >
      customPrefix.second
    </option>
    <option
      value="third"
    >
      customPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests with default value 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value="second"
  >
    <option
      value="default"
    >
      Default
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
</div>
`;

exports[`editable select tests with error 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select has-danger"
>
  <select
    className="custom-select form-control form-control-danger dirty"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
  <div
    className="inline-error"
  >
    error
  </div>
  <i
    className="error-icon buzzn-attention"
  />
</div>
`;

exports[`editable select tests with label 1`] = `
<div
  className="sc-htoDjs joMSuX editable-select false"
>
  <select
    className="custom-select form-control false false"
    id="select"
    name="select"
    value=""
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
  <label
    className="center"
    htmlFor="select"
  />
</div>
`;

exports[`editable select tests with label 2`] = `
<div
  className="sc-htoDjs joMSuX editable-select undefined"
>
  <select
    className="custom-select form-control undefined undefined"
    id="select"
    name="select"
    value="second"
  >
    <option
      value=""
    >
      -----
    </option>
    <option
      value="first"
    >
      objPrefix.first
    </option>
    <option
      value="second"
    >
      objPrefix.second
    </option>
    <option
      value="third"
    >
      objPrefix.third
    </option>
  </select>
  <label
    className="top"
    htmlFor="select"
  />
</div>
`;
